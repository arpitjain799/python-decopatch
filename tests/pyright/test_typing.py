import pytest
import sys
from .base import run_pyright


@pytest.mark.skipif(
    sys.version_info < (3, 7),
    reason="Requires Python 3.7+",
)
def test_typing(snapshot):
    actual = run_pyright("tests/pyright/test_file.py")
    assert actual == snapshot
