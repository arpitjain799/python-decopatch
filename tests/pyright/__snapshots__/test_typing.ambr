# name: test_typing
  <class 'list'> [
    <class 'dict'> {
      'message': '
        No overloads for "function_decorator" match the provided arguments
          Argument types: (Literal[True])
      ',
      'range': <class 'dict'> {
        'end': <class 'dict'> {
          'character': 47,
          'line': 10,
        },
        'start': <class 'dict'> {
          'character': 9,
          'line': 10,
        },
      },
      'rule': 'reportGeneralTypeIssues',
      'severity': 'error',
    },
    <class 'dict'> {
      'message': '
        Argument of type "Literal[2]" cannot be assigned to parameter "scope" of type "str" in function "__call__"
          "Literal[2]" is incompatible with "str"
      ',
      'range': <class 'dict'> {
        'end': <class 'dict'> {
          'character': 26,
          'line': 30,
        },
        'start': <class 'dict'> {
          'character': 25,
          'line': 30,
        },
      },
      'rule': 'reportGeneralTypeIssues',
      'severity': 'error',
    },
  ]
---
